traffic_portal/app/src/common/modules/form/deliveryService/FormDeliveryServiceController.js

```diff
@@ -301,6 +367,46 @@ var FormDeliveryServiceController = function(deliveryService, dsCurrent, origin,
 
     $scope.hasError = formUtils.hasError;
 
+    /**
+     * Checks if a TLS Version has a specific error.
+     *
+     * @param {number} index The index of the TLS Version to check into the
+     * form's Delivery Service's `tlsVersions` array.
+     * @param {string} property The name of the error to check.
+     * @returns {boolean} Whether or not the indicated TLS Version has the given
+     * error.
+     */
+    function tlsVersionHasPropertyError(index, property) {
+        if (!$scope.generalConfig) {
+            return false;
+        }
+        const propName = `tlsVersion${index+1}`;
+        if (!(propName in $scope.generalConfig)) {
+            return false;
+        }
+        return formUtils.hasPropertyError($scope.generalConfig[propName], property);
+    };
+    $scope.tlsVersionHasPropertyError = tlsVersionHasPropertyError;
+
+    /**
+     * Checks if a TLS Version has any error.
+     *
+     * @param {number} index The index of the TLS Version to check into the
+     * form's Delivery Service's `tlsVersions` array.
+     * @returns {boolean} Whether or not the indicated TLS Version has an error.
+     */
+     function tlsVersionHasError(index) {
+        if (!$scope.generalConfig) {
+            return false;
+        }
+        const propName = `tlsVersion${index+1}`;
+        if (!(propName in $scope.generalConfig)) {
+            return false;
+        }
+        return formUtils.hasError($scope.generalConfig[propName]);
+    };
+    $scope.tlsVersionHasError = tlsVersionHasError;
```

Comment:
nit: can be combined with function def

Reply:
It can, but that breaks the JSDoc association, which is the only reason they're separate